pressurecooker/converters.py

```diff
@@ -1,11 +1,80 @@
-import webvtt
+from pycaption import WebVTTReader, SRTReader, SAMIReader, SCCReader, DFXPReader
+from .subtitles import SubtitleConverter, InvalidSubtitleFormatError
+from .subtitles import AmbiguousSubtitleReader, EncapsulatedSubtitleReader
+from le_utils.constants import file_formats
 
 
-def srt2vtt(srt_filename, vtt_filename):
-    try:
-        #webvtt does not accept io.buffered only string path
-        vtt = webvtt.from_srt(srt_filename)
-        #webvtt does not accept io.buffered only string path
-        vtt.save(vtt_filename)
-    except webvtt.errors.MalformedCaptionError as e:
-         return str(e)
+def build_dfxp_reader():
+    return EncapsulatedSubtitleReader(DFXPReader())
+
+
+def build_sami_reader():
+    return EncapsulatedSubtitleReader(SAMIReader())
+
+
+def build_scc_reader():
+    return AmbiguousSubtitleReader(SCCReader())
+
+
+def build_srt_reader():
+    return AmbiguousSubtitleReader(SRTReader())
+
+
+def build_vtt_reader():
+    return AmbiguousSubtitleReader(WebVTTReader())
```

Comment:
@bjester Do you know when we would need a `vtt` reader? 
Is this in case someone calls `convert_subtitles` by mistake even though `in_filename` doesn't need to be converted?

Reply:
The conversion mechanism defaults to detection of the type, which is handled by the reader. So using the reader ensures that VTT file is really a VTT file. I considered adding some switches to escape conversion based on that fact, but in reality it would just add more code and complexity that I don't think is worth it. 

Keeping it also ensures that the output VTT file is formatted correctly and in a standard way they we know we can support downstream, since the WebVTT spec is still in progress. If the reader can't read the file here, then we certainly won't be able read it in Kolibri.